projects/breach-notice-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/WarningService.kt

```diff
@@ -42,19 +41,14 @@ class WarningService(
         val enforceableContacts =
             contactRepository.findEnforceableContacts(disposal!!.event.id)
 
-        val unpaidWorkRequirements = mutableListOf<uk.gov.justice.digital.hmpps.model.Requirement>()
-        repeat(
-            enforceableContacts
-                .filter { upwAppointmentRepository.existsUpwAppointmentsByContactId(it.id) }.size
-        ) {
-            requirementRepository.getUnpaidWorkRequirementsByDisposal(disposal)
-                .forEach { unpaidWorkRequirements.add(it.toModel()) }
-        }
-
         return WarningDetails(
             breachReasons = breachReasons.codedDescriptions(),
             enforceableContacts = enforceableContacts.map(Contact::toEnforceableContact),
-            unpaidWorkRequirements = unpaidWorkRequirements
+            unpaidWorkRequirements = enforceableContacts.filter { it.unpaidWorkAppointments?.isNotEmpty() == true }
+                .let {
+                    requirementRepository.getUnpaidWorkRequirementsByDisposal(disposal.id)
+                        .map { it.toModel() }
+                }
```

Comment:
Not sure if this is doing what you want, the `.let` will always run because `.filter` always returns a non-null list.

I think this would work:
```suggestion
            unpaidWorkRequirements = if (enforceableContacts.any { it.unpaidWorkAppointments.isNotEmpty() == true }) {
                requirementRepository.getUnpaidWorkRequirementsByDisposal(disposal.id)
                    .map { it.toModel() }
            } else emptyList()
```

Reply:
i am still making changes, will raise a pr when i am done

Reply:
was changing the code to use any